internal/cypress/config.go

```diff
@@ -1,316 +1,102 @@
-	Reporters     config.Reporters     `yaml:"reporters,omitempty" json:"-"`
-	Env           map[string]string    `yaml:"env,omitempty" json:"env"`
-	Notifications config.Notifications `yaml:"notifications,omitempty" json:"-"`
-}
-
-// Suite represents the cypress test suite configuration.
-type Suite struct {
-	Name             string        `yaml:"name,omitempty" json:"name"`
-	Browser          string        `yaml:"browser,omitempty" json:"browser"`
-	BrowserVersion   string        `yaml:"browserVersion,omitempty" json:"browserVersion"`
-	PlatformName     string        `yaml:"platformName,omitempty" json:"platformName"`
-	Config           SuiteConfig   `yaml:"config,omitempty" json:"config"`
-	ScreenResolution string        `yaml:"screenResolution,omitempty" json:"screenResolution"`
-	Mode             string        `yaml:"mode,omitempty" json:"-"`
-	Timeout          time.Duration `yaml:"timeout,omitempty" json:"timeout"`
-	Shard            string        `yaml:"shard,omitempty" json:"-"`
-	Headless         bool          `yaml:"headless,omitempty" json:"headless"`
-	PreExec          []string      `yaml:"preExec,omitempty" json:"preExec"`
-	TimeZone         string        `yaml:"timeZone,omitempty" json:"timeZone"`
-}
-
-// SuiteConfig represents the cypress config overrides.
-type SuiteConfig struct {
-	TestFiles         []string          `yaml:"testFiles,omitempty" json:"testFiles"`
-	ExcludedTestFiles []string          `yaml:"excludedTestFiles,omitempty" json:"ignoreTestFiles,omitempty"`
-	Env               map[string]string `yaml:"env,omitempty" json:"env"`
-}
-
-// Reporter represents a cypress report configuration.
-type Reporter struct {
-	Name    string                 `yaml:"name" json:"name"`
-	Options map[string]interface{} `yaml:"options" json:"options"`
-}
-
-// Cypress represents crucial cypress configuration that is required for setting up a project.
-type Cypress struct {
-	// ConfigFile is the path to "cypress.json".
-	ConfigFile string `yaml:"configFile,omitempty" json:"configFile"`
-
-	// Version represents the cypress framework version.
-	Version string `yaml:"version" json:"version"`
-
-	// Record represents the cypress framework record flag.
-	Record bool `yaml:"record" json:"record"`
-
-	// Key represents the cypress framework key flag.
-	Key string `yaml:"key" json:"key"`
-
-	// Reporters represents the customer reporters.
-	Reporters []Reporter `yaml:"reporters" json:"reporters"`
+type Project interface {
+	FilterSuites(suiteName string) error
+	CleanPackages()
+	ApplyFlags(selectedSuite string) error
+	AppendTags([]string)
+	Validate() error
+	IsSharded() bool
+
+	SetDefaults()
+	GetSuiteCount() int
+	GetVersion() string
+	GetRunnerVersion() string
+	SetVersion(version string)
+	GetSauceCfg() config.SauceConfig
+	GetDryRun() bool
+	GetRootDir() string
+	GetSuiteNames() string
+	GetCfgPath() string
+	GetCLIFlags() map[string]interface{}
+	GetArtifactsCfg() config.Artifacts
+	GetShowConsoleLog() bool
```

Comment:
Even though it sounds strange in this specific case, it's more customary to use Has/Is rather than Get for booleans 😅 

```suggestion
	IsShowConsoleLog() bool
```